sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/validator/KeyValidator.java

```diff
@@ -0,0 +1,30 @@
+package org.sdase.commons.keymgmt.validator;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.sdase.commons.keymgmt.KeyMgmtBundle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeyValidator implements ConstraintValidator<PlatformKey, String> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KeyValidator.class);
+  private String keyName;
+
+  @Override
+  public void initialize(PlatformKey constraintAnnotation) {
+    ConstraintValidator.super.initialize(constraintAnnotation);
+    keyName = constraintAnnotation.value();
+  }
+
+  @Override
+  public boolean isValid(String value, ConstraintValidatorContext context) {
+    KeyMgmtBundle<?> keyMgmtBundle = KeyMgmtBundleHolder.getKeyMgmtBundle();
+    if (keyMgmtBundle == null) {
+      LOG.warn("Validation of messages attributes before key-mgmt-bundle is available");
+      return false;
+    } else {
+      return keyMgmtBundle.createKeyManager(keyName).isValidValue(value);
```

Comment:
Seems like [the validation](https://github.com/SDA-SE/sda-dropwizard-commons/blob/718ac0611830513e8b4d908e1492dc58d414512c/sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/manager/KeyManagerImpl.java#L22) does not allow null values. Usually validation annotations like `@Pattern` or `@Email` consider `null` as valid, so that `@NotNull` can be added optionally to the field.

Reply:
Thanks for the hint. Solved this by allowing null values and adding some documentation.